### PR TITLE
Alee/drift radiator draft

### PR DESCRIPTION
**Pull request by andrewrlee**
This is a bit of a WiP

If there is no drift: 
<kbd>
Screenshot 2023-11-22 at 09 20 04
</kbd>

If there is a drift between environments:
<kbd>
Screenshot 2023-11-22 at 09 19 56
</kbd>

Still need to:
* Consider showing multiple radiators for service area / product / team
* Add colours
* Add to table/search?
* etc...

### CODE REVIEW

File: server/views/pages/driftRadiator.njk [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{% extends "../partials/layout.njk" %}
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+{% from "govuk/components/panel/macro.njk" import govukPanel %}
+
+{% set pageTitle = applicationName + " - " + component.name %}
+{% set mainClasses = "app-container govuk-body" %}
+{% set devEnvName = component.environments[0].name %}
```
Comment (andrewrlee) [inline server/views/pages/driftRadiator.njk:7]: assumes the first env in the list represents the head of development

File: server/views/pages/driftRadiator.njk [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{% block content %}
+
+{{ govukBackLink({
+  text: "Back",
+  href: "/components/" + component.name
+}) }}
+
+<h1 id="detailPageTitle">Deployment Drift Radiator</h1>  
+<h2>{{ component.name }}</h2>  
+    <p><a class="govuk-link--no-visited-state" href="https://app.circleci.com/pipelines/github/ministryofjustice/{{ component.name}}?branch=main">Circle CI pipeline</a></p>
```
Comment (andrewrlee) [inline server/views/pages/driftRadiator.njk:18]: Assumes all services are on circle - I was hoping to link through to builds on each tile but as well as build number, also need workflow id, which isn't available

File: assets/js/driftRadiator.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  if (env != devEnvName && data[devEnvName]) {
+    const devEnvSha = data[devEnvName].sha
+    const thisEnvSha = data[env].sha
+
+    const devEnvDate = data[devEnvName].date
+    const thisEnvDate = data[env].date
+
+    const humanReadableDate = `${dayjs(devEnvDate).diff(thisEnvDate, 'day')} days behind dev`
+    const gitDiffUrl = `https://github.com/ministryofjustice/${componentName}/compare/${thisEnvSha}...${devEnvSha}`
```
Comment (andrewrlee) [inline assets/js/driftRadiator.js:26]: This allows you to see the difference in commits between this env and dev in github, e.g: [link](https://github.com/ministryofjustice/create-and-vary-a-licence-api/compare/f35bf67...fc8a524)

This provides one of the benefits of the deployment tool - allows you to work out what tickets are inflight.
(The other bit of the deployment tool is pulling out jira ticket ids and displaying their current ticket state)

File: server/views/pages/driftRadiator.njk (hunk 2 above)
Comment (mattops) [reply server/views/pages/driftRadiator.njk:18]: the old monitor called the circleci API to get the latest build status etc.

Comment (andrewrlee) [reply server/views/pages/driftRadiator.njk:18]: it's a shame we don't have the pipeline id available as part of the build number (as well as the job number) 

Then we could link to this page:
https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-developer-portal/544

Comment (andrewrlee) [reply server/views/pages/driftRadiator.njk:18]: there's [an api](https://circleci.com/docs/api/v2/index.html#operation/getJobDetails) that grabs the pipeline id for job.

You could then grab the pipeline and build up any url.

Which would support:
"As a user, I want to easily progress a known deployment from one environment to the next"

**Pull request merged**